Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -32,6 +32,7 @@ extension PackageGraph {
         customPlatformsRegistry: PlatformRegistry? = .none,
         customXCTestMinimumDeploymentTargets: [PackageModel.Platform: PlatformVersion]? = .none,
         testEntryPointPath: AbsolutePath? = nil,
+        availableLibraries: [LibraryMetadata],
```

Comment:
Can we have a default empty array argument here so that clients outside of this package relying on this function don't break? This would also fix the smoke test CI jobs, as SourceKit-LSP can't be built with this PR as is.

Reply:
I think it's correct that this is a breaking change? 

I would see it similar to the `binaryArtifacts` parameter which also doesn't offer a default value. Whatever gets passed in *has* to match what was provided to the underlying workspace, so it's important that clients make a deliberate choice here. 

I think in the fullness of time this probably shouldn't be a public API, though, or it should be refactored in a way that parameters that are associated with the workspace can't be arbitrarily passed here.

Reply:
Oh, this is the `loadPackageGraph()` on the workspace. 

Yah it's a little unfortunate that it needs the parameter, but it's basically the consequence of the toolchain being private and that code being in an extension of workspace. I didn't really want to change the existing assumption that the toolchain object wouldn't be usable there.